easybuild/tools/module_generator.py

```diff
@@ -998,6 +1009,15 @@ def __init__(self, *args, **kwargs):
             if self.modules_tool.version and LooseVersion(self.modules_tool.version) >= LooseVersion('7.7.38'):
                 self.DOT_MODULERC = '.modulerc.lua'
 
+    def check_version(self, minimal_version_maj, minimal_version_min):
```

Comment:
You should also add this method to the `ModuleGenerator` class, and make it `raise NotImplementedError` there in case someone trigger it for `ModuleGeneratorTcl`?

Either that, or rename to something with a leading underscore to make it "private", like `_check_lmod_version`